akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala

```diff
@@ -262,8 +262,9 @@ object RejectionHandler {
       }
       .handleAll[UnsupportedRequestContentTypeRejection] { rejections =>
         val supported = rejections.flatMap(_.supported).mkString(" or ")
-        val unsupportedContentType = rejections.find(_.contentType.isDefined).map(_.contentType)
-        rejectRequestEntityAndComplete((UnsupportedMediaType, s"The request's Content-Type [$unsupportedContentType] is not supported. Expected:\n" + supported))
+        val unsupportedContentType =
+          rejections.find(_.contentType.isDefined).flatMap(_.contentType).fold("")(" [" + _ + "]")
```

Comment:
Perhaps this could be written shorter, something like `rejections.flatMap(_.contentType).mkString(" [", ", ", "]")` perhaps?

Reply:
It's an `Option` so that wouldn't give empty string for `None` though

Reply:
aah right I got confused